interfaces/GithubFactory.py

```diff
@@ -223,12 +224,22 @@ def create_test_repo(self, shaBase, repoFullName: str, path_repos: str):
         
         Parameters:
             shaBase : The commit hash of the base commit to check out in the test repository."""
+        self.previousSha = 0
         if not os.path.exists(path_repos):
             if not os.path.exists("./test"):
                 os.mkdir("./test")
             os.system(f"cd ./test && git clone https://github.com/{repoFullName}")
         
         os.system(f"cd {path_repos} && git checkout {shaBase}")
+        if self.previousSha == 0:
+            self.previousSha = shaBase
+            return None
+        else:
```

Comment:
the function logic no longer match the function name. Maybe this function has to be split into 2 functions